app/src/main/java/com/epfl/drawyourpath/userProfileCreation/PhotoProfileInitFragment.kt

```diff
@@ -0,0 +1,138 @@
+package com.epfl.drawyourpath.userProfileCreation
+
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.ImageDecoder
+import android.os.Bundle
+import android.provider.MediaStore
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.epfl.drawyourpath.R
+import com.epfl.drawyourpath.authentication.FirebaseAuth
+import com.epfl.drawyourpath.authentication.MockAuth
+import com.epfl.drawyourpath.database.Database
+import com.epfl.drawyourpath.database.FireDatabase
+import com.epfl.drawyourpath.database.MockDataBase
+import com.epfl.drawyourpath.login.LoginActivity
+import java.util.concurrent.CompletableFuture
+
+class PhotoProfileInitFragment : Fragment(R.layout.fragment_photo_profile_init) {
+    private var isTest: Boolean = false
+    private var username: String = ""
+
+    private var photoProfile: Bitmap? = null
+    private val requestCodeFrag = 100
+
+    private lateinit var imageView: ImageView
+    private lateinit var errorText: TextView
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        var database: Database = FireDatabase()
+
+        //retrieve the isRunTestValue and userName from the PersonalInfoFragment
+        val argsFromLastFrag: Bundle? = arguments
+        if (argsFromLastFrag == null) {
+            isTest = false
+        } else {
+            isTest = argsFromLastFrag.getBoolean("isRunningTestForDataBase")
+            username = argsFromLastFrag.getString(database.usernameFile).toString()
+        }
+
+        //select the correct database in function of test scenario
+        database = if (isTest) {
+            MockDataBase()
+        } else {
+            FireDatabase()
+        }
+
+        imageView= view.findViewById(R.id.imagePhotoProfileInitFrag)
+        errorText = view.findViewById(R.id.photoProfile_error_text)
+
+        val selectPhotoButton: Button = view.findViewById(R.id.selectPhotoInitPhotoFrag)
+        selectPhotoButton.setOnClickListener{
+            val photoPicker = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI)
+            startActivityForResult(photoPicker, requestCodeFrag)
+        }
+
+        //if we click on the skip button we simply pass the end profile creation fragment
+        val skipButton: Button = view.findViewById(R.id.skipPhotoProfile_button_userProfileCreation)
+        skipButton.setOnClickListener{
+            showEndProfileCreationFrag(database)
+        }
```

Comment:
could be moved to a private function
`private fun createSkipButton(view: View, database: Database)`